### DISCUSSION TITLE
类式组件调用useAccess报错咋整....

### DISCUSSION THREAD

**Discussion by Jixiangup** [umijs/umi#7389] [Q&A]
这是报错信息:
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

源码：
`import React, {Component} from 'react';
import { useAccess, Access } from 'umi';

interface Props {
    haha: string
}

interface State {

}

export default class AccessTest extends Component<Props, State> {
    render() {

        console.log("haha", this.props.haha);
        

        const access = useAccess()
        console.log("access", access)
        return (
            <div>
               
            </div>
        )
    }
}`
在插件GitHub仓库找到有人提出这个issues，但是没人回答。。

**Comment by ingdaeho** [comment]
you should use functional component, not class component
